File: batches_cpp/batch_04_comment_addition/ref.cpp
#include <iostream>
#include <vector>
using namespace std;

int findMax(const vector<int>& arr) {
    if (arr.empty()) return INT_MIN;
    int maxVal = arr[0];
    for (int i = 1; i < (int)arr.size(); i++) {
        if (arr[i] > maxVal) maxVal = arr[i];
    }
    return maxVal;
}

int main() {
    vector<int> data = {3, 7, 2, 9, 1};
    cout << findMax(data) << endl;
    return 0;
}

File: batches/batches/batch_04_comment_addition/ref.py
"""
Reference - Original code for Batch 4: Comment Addition
Finds maximum in a list.
"""


def find_max(arr):
    if not arr:
        return None
    max_val = arr[0]
    for x in arr[1:]:
        if x > max_val:
            max_val = x
    return max_val


if __name__ == "__main__":
    data = [3, 7, 2, 9, 1]
    print(find_max(data))
